File: src/types/broker.ts
export interface BrokerHolding {
  symbol: string
  quantity: number
  average_price: number
  last_price: number
  pnl: number
}

export interface BrokerAccountSummary {
  provider: string
  account_id: string | null
  account_name: string | null
  total_investment: number
  current_value: number
  available_cash: number
  pnl: number
  holdings: BrokerHolding[]
  status: string
  mode: string
  message: string
}

File: backend/schemas/broker.py
from pydantic import BaseModel, Field


class BrokerHolding(BaseModel):
    symbol: str
    quantity: float
    average_price: float
    last_price: float
    pnl: float


class BrokerAccountSummary(BaseModel):
    provider: str
    account_id: str | None = None
    account_name: str | None = None
    total_investment: float = 0.0
    current_value: float = 0.0
    available_cash: float = 0.0
    pnl: float = 0.0
    holdings: list[BrokerHolding] = Field(default_factory=list)
    status: str
    mode: str
    message: str
